cmd/erasure-server-pool.go

```diff
@@ -582,6 +582,15 @@ func (z *erasureServerPools) NSScanner(ctx context.Context, bf *bloomFilter, upd
 func (z *erasureServerPools) MakeBucketWithLocation(ctx context.Context, bucket string, opts BucketOptions) error {
 	g := errgroup.WithNErrs(len(z.serverPools))
 
+	// Lock the bucket name before creating.
+	lk := z.NewNSLock(bucket, "/MakeBucketWithLocation")
```

Comment:
This can block with an object name that can be potentially named as `MakeBucketWithLocation` it is better to do this locking on a reserved bucket name

```suggestion
	lk := z.NewNSLock(minioMetaTmpBucket, bucket+".lck")
```